File: CS/WindowsApplication1/CustomEditor/RepositoryItemMyHyperlinkEdit.cs
using System;
using System.ComponentModel;
using System.Drawing;
using System.Windows.Forms;
using DevExpress.XtraEditors;
using DevExpress.XtraEditors.Repository;
using DevExpress.XtraEditors.Registrator;
using DevExpress.XtraEditors.ViewInfo;
using DevExpress.Utils.Drawing;

namespace WindowsApplication1 {
    [System.ComponentModel.DesignerCategory("")]
	[UserRepositoryItem("Register")]
	public class RepositoryItemMyHyperlinkEdit : RepositoryItemHyperLinkEdit { 
		static RepositoryItemMyHyperlinkEdit() { 
			Register(); 
		}
		public RepositoryItemMyHyperlinkEdit() {}
		
		internal const string EditorName = "MyHyperlinkEdit";

		public static void Register() {
			EditorRegistrationInfo.Default.Editors.Add(new EditorClassInfo(EditorName, typeof(MyHyperlinkEdit),
                typeof(RepositoryItemMyHyperlinkEdit), typeof(MyHyperLinkEditViewInfo),
                new MyHyperLinkEditPainter(), true));
		}
		public override string EditorTypeName { 
			get { return EditorName; } 
		}
	}
}

File: VB/WindowsApplication1/CustomEditor/RepositoryItemMyHyperlinkEdit.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.ComponentModel
Imports System.Drawing
Imports System.Windows.Forms
Imports DevExpress.XtraEditors
Imports DevExpress.XtraEditors.Repository
Imports DevExpress.XtraEditors.Registrator
Imports DevExpress.XtraEditors.ViewInfo
Imports DevExpress.Utils.Drawing

Namespace WindowsApplication1
	<System.ComponentModel.DesignerCategory(""), UserRepositoryItem("Register")> _
	Public Class RepositoryItemMyHyperlinkEdit
		Inherits RepositoryItemHyperLinkEdit
		Shared Sub New()
			Register()
		End Sub
		Public Sub New()
		End Sub

		Friend Const EditorName As String = "MyHyperlinkEdit"

		Public Shared Sub Register()
			EditorRegistrationInfo.Default.Editors.Add(New EditorClassInfo(EditorName, GetType(MyHyperlinkEdit), GetType(RepositoryItemMyHyperlinkEdit), GetType(MyHyperLinkEditViewInfo), New MyHyperLinkEditPainter(), True))
		End Sub
		Public Overrides ReadOnly Property EditorTypeName() As String
			Get
				Return EditorName
			End Get
		End Property
	End Class
End Namespace